### PR TITLE
Full update

### PR DESCRIPTION
**Pull request by pasinskim**

### CODE REVIEW

File: daemon.go [hunk 1]
```diff
@@ -61,28 +70,43 @@ func getMenderServer(serverFile string) string {
...
-	updater     updateRequester
-	config      daemonConfigType
+	//updater     updateRequester
+	client *Client
+	daemonConfigType
 	stopChannel chan (bool)
 }
 
-func (daemon menderDaemon) quitDaaemon() {
+func (daemon menderDaemon) StopDaaemon() {
```
Comment (kacf) [inline daemon.go]: `s/StopDaaemon/StopDaemon/`

File: update.go [hunk 2]
```diff
@@ -71,67 +39,62 @@ type updateHaveUpdateResponseType struct {
...
+
+	var update UpdateResponse
+	err = ProcessUpdateResponse(r, &update)
+
+	if err != nil {
+		return "", err
 	}
-	// ureachable
+
+	if r.StatusCode == updateRespponseHaveUpdate {
```
Comment (kacf) [inline update.go]: Preexisting, but still: `s/Respponse/Response/`

**Comment by kacf** [comment]
Nice cleanup!

**Comment by pasinskim** [comment]
@kacf @maciejmrowiec some more improvements and cleanup of the client code. I will post some dependency diagram later today to make this clear, but there is nice description in the commit message: https://github.com/mendersoftware/mender/pull/38/commits/ec16aa6cce7e15b4250acc81ff1080801f36d9d2
Test coverage is up, but some more tests are coming.

File: bootenv.go [hunk 3]
```diff
@@ -16,23 +16,50 @@ package main
...
 	"strings"
+
+	"github.com/mendersoftware/log"
 )
 
-type uBootEnvCommand struct {
-	EnvCmd string
+type BootVars map[string]string
+
+type BootEnvReadWritter interface {
```
Comment (kacf) [inline bootenv.go]: `s/Writter/Writer/` throughout.

File: ioctl.go [hunk 4]
```diff
@@ -24,10 +27,12 @@ import "unsafe"
 // instead.
 type ioctlRequestValue uintptr
 
+var NotABlockDevice = errors.New("Not a block device.")
+
 // Returns size in first return. Second returns true if descriptor is not a
 // block device. If it's true, then error != nil. Last return is error
 // condition.
-func getBlockDeviceSize(file *os.File) (uint64, bool, error) {
+func getBlockDeviceSize(file *os.File) (uint64, error) {
```
Comment (kacf) [inline ioctl.go]: With your change, the comment is now wrong, it should be changed as well.

**Comment by kacf** [comment]
This is way too big to go in deep detail on everything, but it looks good from an overview perspective, I've added a few comments here and there. You're right that it's good that we have large test coverage for this. I like the module based splitting!

**Comment by pasinskim** [comment]
@kacf can you run your integration test against this one? If those are passing maybe we can merge this?

**Comment by kacf** [comment]
Sorry, I'm having build errors in poky, and I'm deep in CFEngine right now. You should be able to run it yourself, just follow the README in `tests/acceptance`.

**Pull request merged**